api/base/serializers.py

```diff
@@ -345,8 +397,10 @@ def to_representation(self, obj, envelope='data'):
             except SkipField:
                 continue
 
-            if isinstance(field, JSONAPIHyperlinkedIdentityField):
+            if isinstance(field, (ser.HyperlinkedRelatedField, RelationshipField)):
                 data['relationships'][field.field_name] = field.to_representation(attribute)
+                if data['relationships'][field.field_name] == {'links': {}}:
```

Comment:
This feels a bit hacky. Could you make it so that `RelationshipField` serializes to `None` so that you can do

```python
result = field.to_representation(attribute)
if result:
    if data['relationships'][field.field_name] = result
```

here?